internal/extension/manager.go

```diff
@@ -141,19 +145,44 @@ func newExtensionService(dag *nilGuardedPointer[StateAwareDAG]) extpb.ExtensionS
 }
 
 func (s *extensionService) Subscribe(_ *emptypb.Empty, stream grpc.ServerStreamingServer[extpb.Event]) error {
+	channel := BlockingDAG.newUpdateChannel()
 	for {
-		time.Sleep(time.Second * 5)
-		if err := stream.Send(&extpb.Event{}); err != nil {
-			return err
+		dag := <-channel
+		opts := []cel.EnvOption{
+			kuadrant.CelExt(&dag),
 		}
+
+		s.mutex.Lock()
```

Comment:
using defer would also be an option for the unlock

Reply:
I think this doesn't work here because defer won't run as it's in an infinite for loop? would only work if it was wrapped in an anonymous func iiuc

Reply:
My IDE seemed to have disagreed with you, cause that's what I wrote initially. I know I have to educate myself. But I think it is related to it being in a `for` loop? Is `defer` postponing to lexical scope ending, or function ending? I _think_ it's the former, i.e. pushing it onto the stack, which here wouldn't be what we want, right?